src/scripts/main.js

```diff
@@ -1,3 +1,212 @@
 'use strict';
 
-// write code here
+const headers = [...document.querySelector('thead').firstElementChild.children];
+const tbody = document.querySelector('tbody');
+let people = [...document.querySelector('tbody').children];
+
+headers.forEach(header => {
+  header.setAttribute('Order', 'asc');
+
+  header.addEventListener('click', (e) => {
+    const copyHeaders = [...headers];
+
+    copyHeaders.splice(e.target.cellIndex, 1);
+    copyHeaders.forEach(head => head.setAttribute('Order', 'asc'));
+
+    people.sort((person1, person2) => {
+      let per1 = person1.children[e.target.cellIndex].textContent;
+      let per2 = person2.children[e.target.cellIndex].textContent;
+
+      if (header.getAttribute('Order') === 'dsc') {
+        per1 = per2;
+        per2 = person1.children[e.target.cellIndex].textContent;
+      }
+
+      if (e.target.cellIndex === 4) {
+        per1 = +per1.slice(1).replaceAll(',', '');
+        per2 = +per2.slice(1).replaceAll(',', '');
+
+        return per1 - per2;
+      };
+
+      return per1.localeCompare(per2);
+    });
+
+    header.getAttribute('Order') === 'asc'
+      ? header.setAttribute('Order', 'dsc')
+      : header.setAttribute('Order', 'asc');
+
+    tbody.append(...people);
+  });
+});
+
+people.forEach(person => {
+  person.addEventListener('click', (e) => {
+    const copyPeople = [...people];
+
+    copyPeople.splice(e.target.cellIndex, 0);
+    copyPeople.forEach(employee => employee.classList.remove('active'));
+
+    person.classList.add('active');
+  });
+});
+
+document.body.insertAdjacentHTML('beforeend', `
+  <form class="new-employee-form" action="/" method="GET">
+    <label>
+      Name: <input name="name" type="text"  data-qa="name">
+    </label>
+
+    <label>
+      Position: <input name="position" type="text" data-qa="position">
+    </label>
+
+    <label>
+      Office: <select name="office" data-qa="office">
+        <option>Tokyo</option>
+        <option>Singapore</option>
+        <option>London</option>
+        <option>New York</option>
+        <option>Edinburgh</option>
+        <option>San Francisco</option>
+      </select>
+    </label>
+
+    <label>
+      Age: <input name="age" data-qa="age" type="number">
+    </label>
+
+    <label>
+      Salary: <input name="salary" type="number" data-qa="salary">
+    </label>
+
+    <button type="submit">Save to table</button>
+  </form>
+`);
+
+const form = document.querySelector('form');
```

Comment:
Consider using a more specific selector for the form, like `document.querySelector('.new-employee-form')`, to avoid accidentally selecting the wrong form.

Reply:
done